pkg/sfu/downtrackspreader.go

```diff
@@ -102,11 +102,13 @@ func (d *DownTrackSpreader) HasDownTrack(peerID livekit.ParticipantID) bool {
 
 func (d *DownTrackSpreader) Broadcast(layer int32, pkt *buffer.ExtPacket) {
 	d.downTrackMu.RLock()
-	downTracks := d.downTracks
-	free := d.free
+	var downTracks []TrackSender
+	downTracks = append(downTracks, d.downTracks...)
```

Comment:
Don't like this one as this happens on every packet. May not be too bad as it is just copying a bunch of pointers, but would be good to avoid if possible.

We do check that the element in the slice is not `nil` before using it. But, it is possible that some element is set to `nil` in the underlying array when a down track is deleted. So, the race checker is catching that and reporting it. A copy is the way to fix it, but maybe not required and we can just ignore the race report as an understood issue.

Reply:
There are in fact APIs (`GetDownTracks`) which just returns the slice. So, in theory the call sites could hit the race too which would need a return from those APIs a copied slice.

Reply:
agreed this would be good to avoid when it happens on every packet. if we are using it in an race compliant way.. can we somehow mark to the race detector to ignore that warning

Reply:
Will check if there is a way.

Reply:
There is this option to exclude race check - https://go.dev/doc/articles/race_detector#Excluding_Tests. I am trying to place it around only sections of code where we know it is safe. But, as soon as I put it in the code in that spot, `gofmt` moves it to the top of the file (maybe just a gofmt thing) which would mean the entire file is excluded. Maybe, we need to put functions which we are know are race safe in a separate file? That would make things ugly, but maybe the only way to do it :-(

Reply:
this is a file-specific attribute. It's not possible to only exclude part of a file. If it's desired to exclude a particular function, then you can move that function into a new file.

Reply:
Got it. Yeah, that's what I am looking at it now.

Reply:
@davidzhao @cnderrauber I am going to make another commit to remove this copy for now. We know it is race safe. We will probably not need to take race report at face value and check on a case-by-case basis.